docs/Coiote_IoT_DM/firmware_update.md

```diff
@@ -0,0 +1,188 @@
+
+        - `CoAP` or `CoAPs` over `UDP`
+        - `CoAP` or `CoAPs` over `TCP`
+        - `HTTP` or `HTTPs`
+
+        **Push** uses `CoAPs` over `UDP` by default.
+
+    !!! Note
+        **Which transport protocol to choose?**
+        
+        Downloads using `CoAP(s)` over `UDP` tend to be slow due to limitation of the maximum CoAP Block size of 1024 bytes and the required acknowledgements for each Block transfer.
+
+        Choosing `CoAP(s)` over `TCP` or `HTTP(s)` usually results in faster download speeds.
+
+
+    ![Basic Firmware Update](images/Pull.png)
+
+
+1. Click **Schedule Update** from the bottom-right corner to trigger the Firmware Update process.
+
+    ![Schedule Update](images/schedule.png)
+
+
+## Download & Upgrade Process
+
+If the Firmware Update is scheduled successfully, the device starts **downloading** the firmware at the next practical opportunity. The actual firmware **update** starts once the integrity and authenticity of the firmware image has been validated by the LwM2M Client.
+
+![Upgrading](images/update-in-progress.png)
+
+The status of the firmware update is shown in the Resources: **State** `/5/*/3` & **Update Result** `/5/*/5`. If no errors arise, the update process follows the following pattern:
+
+1. **Downloading** `state 1` & `update result 0` 
+2. **Downloaded** `state 2` & `update result 0` 
+3. **Updating** `state 3` & `update result 0` 
+4. **Updated** `state 0` & `update result 1` 
+
+
+!!! info
+    When the device is updating its firmware, it will deregister and reboot using the new firmware. This process may time multiple minutes.
+
+    ![Deregistered while upgrading](images/deregistered.png)
+
+
+If the update was successfully performed, the **State** `/5/*/3` returns to `0` and the **Update Results** `/5/*/5` reports `1`.
+
+![Firmware upgraded](images/result1.png)
+
+The **Update list** also reports: `Success`.
+
+![Successful update](images/success.png)
+
+
+
+## Troubleshooting
+
+* ### Insufficient flash
+
+    A common update error is due to insufficient flash memory, indicated by **Update Result** `2`.
+
+    The Anjay build file can be quite large in size, particularly when using the `.hex` file (`app_signed.hex`). For updating the firmware, it’s better to use the `app_update.bin` file which is about half the size of the `.hex` file.
+
+    !!! Warning
+        The **Nordic Thingy:91** is very restricted in its flash size due to the Nordic Bootloader. The maximum firmware size is `421kB`.
+
+* ### Firmware Update only works over CoAP, not over CoAPs
+
+    !!! Warning
+        Possible issue when working with the **Zephyr LwM2M Client**.
+    
+    Firmware Update over CoAPs may requires some configuration in **Kconfig**.
+
+    The **security tags** needs to be similar in the "**Security tag for FOTA download library**" as in the "**LwM2M server TLS tag**". Both tags need to be set to either `16842753` or `35724861`.
```

Comment:
Firstly - not "similar". They need to be exactly the same value.

Secondly - the numeric values don't actually matter. The numbers you're quoting are just defaults used by Nordic's provisioning scripts and what's provisioned onto the modem at the factory - so yes, that's going to be easiest to use for most people. There's, however, otherwise nothing special about those and you can provision credentials with any ID from code, by using AT commands, or with the [Certificate Manager feature of nRF Connect LTE Link Manager](https://infocenter.nordicsemi.com/topic/ug_link_monitor/UG/link_monitor/lm_certificate_manager.html)). So the "both need to be set to..." wording might be misleading.